src/components/ArticleVotes.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import propTypes from "prop-types";
+// import ErrorMessage from "./ErrorMessage";
+// import Loading from "./Loading";
+import { patchData } from "./api";
+
+function ArticleVotes({ article_id, article_votes }) {
+  const [voteCounter, setVoteCounter] = useState(0);
+
+  const handleClick = (vote) => {
+    setVoteCounter((currVotes) => currVotes + vote);
+    patchData(article_id, vote);
```

Comment:
make sure to undo the optimistic rendering if the request fails and the vote doesnt get added to the db